test/easyconfigs/easyconfigs.py

```diff
@@ -394,6 +394,18 @@ def check_dep_vars(self, gen, dep, dep_vars):
                     if all(ec.startswith('%s-%s-' % (parent_name, dep_ver)) for ec in ecs) and len(dep_vars) > 1:
                         dep_vars.pop(key)
 
+        # multiple variants of Meson is OK as long as they are deps for meson-python, since meson-python should only be
+        # a build dependency elsewhere
+        if dep == 'Meson' and len(dep_vars) > 1:
+            for key in list(dep_vars):
+                ecs = dep_vars[key]
+                # filter out Meson variants that are only used as a dependency for meson-python
+                if all(ec.startswith('meson-python-') for ec in ecs):
+                    dep_vars.pop(key)
+                # always retain at least one dep variant
+                    if len(dep_vars) == 1:
+                        break
```

Comment:
Not sure if the `if` statement should be inside the other `if` or not, but either the comment or the `if` statement doesn't have the correct indentation here.

Reply:
good spot
functionally, I don't think it matters whether it's indented or not, but probably makes more sense to not indent